easyrelease-lib/src/main/groovy/eu/inloop/easyrelease/plugin/Util.groovy

```diff
@@ -0,0 +1,142 @@
+package eu.inloop.easyrelease.plugin
+import groovy.text.SimpleTemplateEngine
+import org.ajoberstar.grgit.Grgit
+
+class Util {
+
+    static final String TAG = '[easyrelease]'
+
+    static final String DEFAULT_APK_STRUCTURE = '$projectName-$variantName-$variantVersionName-$variantVersionCode'
+
+    /**
+     * Modifies output apk file name for all application variants
+     */
+    static def setApkName(project) {
+        project.android.applicationVariants.all { variant ->
+
+            def template = getNameString(project)
+            def binding = getBinding(project, variant)
+
+            def engine = new SimpleTemplateEngine()
+            def fileName = engine.createTemplate(template).make(binding).toString() + ".apk"
+
+            try {
+                variant.outputs.each { output ->
+                    output.outputFile = new File(output.outputFile.parent, fileName)
+                    println "$TAG Setting $output.name variant output name to $fileName"
+                }
+            } catch (all) {
+                println "$TAG Exception raised during build: " + all
+            }
+        }
+    }
+
+    static def getNameString(project) {
+        def projectDir = project.projectDir
+        def propFileName = 'easyrelease.properties'
+        def propFile = new File("$projectDir/$propFileName")
+        if (propFile.canRead()) {
+            def props = new Properties()
+            props.load(new FileInputStream(propFile))
+
+            if (props["APK_NAME"] != null) {
+                return props["APK_NAME"]
+            }
+            println "$TAG APK_NAME is null, using the default one "
+        }
+        return DEFAULT_APK_STRUCTURE
+    }
+
+    static def getBinding(project, variant) {
+        def now = Calendar.getInstance()
+        def binding = [
+                projectName : project.name,
+                variantName : variant.name,
+                // first check version code and name from Gradle build script, then from AndroidManifest.xml
+                variantVersionCode : variant.versionCode ? variant.versionCode : getVersionCode(project),
+                variantVersionName : variant.versionName ? variant.versionName : getVersionName(project),
+                dateYear : now.get(Calendar.YEAR),
+                dateMonth : now.get(Calendar.MONTH) + 1,
+                dateDay : now.get(Calendar.DAY_OF_MONTH),
+                dateHour : now.get(Calendar.HOUR_OF_DAY),
+                dateMinute : now.get(Calendar.MINUTE),
+                dateSecond : now.get(Calendar.SECOND),
+                dateMillisecond : now.get(Calendar.MILLISECOND)
+        ]
+
+        try {
+            def repo = Grgit.open(project.file('..'))
```

Comment:
Try to research/try whether its possible to rely on traditional git lookup, which usualy searches the closest .git from current directory to root directory.